eventuate-adapter-vertx/src/main/scala/com/rbmhtechnology/eventuate/adapter/vertx/api/EventMetadata.scala

```diff
@@ -0,0 +1,110 @@
+    val LocalSequenceNr = "localSequenceNr"
+    val LocalLogId = "localLogId"
+    val EmitterId = "emitterId"
+  }
+
+  // exposure through a value to make the headers accessible for the Java-API
+  val Headers = Headers_
+
+  /**
+   * Creates an instance of [[EventMetadata]] containing event metadata extracted from the headers of an event bus message
+   * if this message originated from a Vert.x producer.
+   *
+   * @param headers Headers of the event bus message.
+   * @return An [[Option]] containing the metadata if this message was sent by a Vert.x producer otherwise [[None]].
+   */
+  def apply(headers: MultiMap): Option[EventMetadata] =
+    headers.getAsOption(MessageProducer).filter(_ == VertxProducer)
+      .map { s =>
+        new EventMetadata(
+          headers.getOrElseThrow(Headers.LocalLogId),
+          headers.getOrElseThrow(Headers.LocalSequenceNr).toLong,
+          headers.getOrElseThrow(Headers.EmitterId))
+      }
+
+  /**
+   * Creates an instance of [[EventMetadata]] containing event metadata extracted from the headers of an event bus message
+   * if this message originated from a Vert.x producer.
+   *
+   * @param headers Headers of the event bus message.
+   * @return An [[Option]] containing the metadata if this message was sent by a Vert.x producer otherwise [[None]].
+   */
+  def apply(headers: RxMultiMap): Option[EventMetadata] =
+    apply(headers.getDelegate.asInstanceOf[MultiMap])
+
+  /**
+   * Java-API that creates an instance of [[EventMetadata]] containing event metadata extracted from the headers of an event bus message
+   * if this message originated from a Vert.x producer.
+   *
+   * @param headers Headers of the event bus message.
+   * @return An [[JOption]] containing the metadata if this message was sent by a Vert.x producer otherwise an empty [[JOption]].
+   */
+  def fromHeaders(headers: MultiMap): JOption[EventMetadata] =
+    apply(headers).asJava
+
+  /**
+   * Java-API that creates an instance of [[EventMetadata]] containing event metadata extracted from the headers of an event bus message
+   * if this message originated from a Vert.x producer.
+   *
+   * @param headers Headers of the event bus message.
+   * @return An [[JOption]] containing the metadata if this message was sent by a Vert.x producer otherwise an empty [[JOption]].
+   */
+  def fromHeaders(headers: RxMultiMap): JOption[EventMetadata] =
+    apply(headers).asJava
+
+  private[vertx] def apply(event: DurableEvent): EventMetadata =
+    new EventMetadata(event.localLogId, event.localSequenceNr, event.emitterId)
+}
+
+/**
+ * Contains metadata of an event originating from an event log.
+ *
+ * @param localLogId      Id of the local event log.
+ * @param localSequenceNr Sequence number in the local event log.
+ * @param emitterId       Id of the emitter ([[EventsourcedActor]] or [[EventsourcedProcessor]]).
+ */
+class EventMetadata(val localLogId: String, val localSequenceNr: Long, val emitterId: String) {
```

Comment:
Both, Java and Scala applications now use `EventMetadata`from the `api` package. OTOH the config part of the API is strictly separated into different packages but here you make an exception. 

When you move `Metadata` one level up, you have one part of the API in the `vert.x` package and the other part in the `(j)api` package. What about renaming `(j)api` to `(j)config`?